internal/xds/translator/route.go

```diff
@@ -440,8 +440,20 @@ func buildXdsURLRewriteAction(destName string, urlRewrite *ir.URLRewrite, pathMa
 	return routeAction
 }
 
-func buildXdsDirectResponseAction(res *ir.DirectResponse) *routev3.DirectResponseAction {
-	routeAction := &routev3.DirectResponseAction{Status: res.StatusCode}
+func buildXdsDirectResponseAction(res *ir.CustomResponse) *routev3.DirectResponseAction {
```

Comment:
What if `res.StatusCode == nil && (res.Body == nil || res.Body == "")`?

Reply:
in the gateway api layer we are always setting the status code, so we should not be hitting this case

Reply:
Oh I see, the default has been set to 200 in gateway translation.

It would be more clearer if a default 200 is also set in the xds translation.